src/d/actor/d_a_ff.cpp

```diff
@@ -1,55 +1,363 @@
+        g_dComIfG_gameInfo.drawlist.setOpaListMaskOff();
+        g_dComIfG_gameInfo.drawlist.setXluListMaskOff();
+        mDoExt_modelUpdate(i_this->mpModel[0]);
+        if (i_this->mGlowScale > 0.01f) {
+            mDoMtx_YrotM(*calc_mtx, i_this->mLiveTimer << 8);
+            MtxScale(i_this->mGlowScale, i_this->mGlowScale * i_this->mGlowScaleY,
+                     i_this->mGlowScale, true);
+            i_this->mpModel[1]->setBaseTRMtx(*calc_mtx);
+            i_this->mBrkAnm[1]->play();
+            i_this->mBrkAnm[1]->entry(i_this->mpModel[1]->getModelData());
+            mDoExt_modelUpdate(i_this->mpModel[1]);
+        }
+        g_dComIfG_gameInfo.drawlist.setOpaList();
+        g_dComIfG_gameInfo.drawlist.setXluList();
+    }
+    return;
 }
 
 /* 000001E4-000002EC       .text z_check__FP8ff_classP4cXyz */
-void z_check(ff_class*, cXyz*) {
-    /* Nonmatching */
+static void z_check(ff_class* i_this, cXyz* param_2) {
+    u32 local_28;
+
+    if ((param_2->x < 0.0f) || (param_2->x > 638.99f) || (param_2->y < 0.0f) ||
+        (param_2->y > 526.99f) || (param_2->z >= 0.0f))
+    {
+        i_this->mbNotVisibleZ = 1;
+    } else {
+        GXDrawDone();
+        GXPeekZ((u16)param_2->x, (u16)param_2->y, &local_28);
+        f32 fVar1 = 0.5f - param_2->z * -1.0f;
+        if (fVar1 < 0.0f) {
+            fVar1 = 0.0f;
+        }
+        if ((u32)(fVar1 * 33553002.0f) > local_28) {
+            i_this->mbNotVisibleZ = 1;
+        } else {
+            i_this->mbNotVisibleZ = 0;
+        }
+    }
+    return;
 }
 
 /* 000002EC-0000037C       .text daFf_Draw__FP8ff_class */
-static BOOL daFf_Draw(ff_class*) {
-    /* Nonmatching */
+static BOOL daFf_Draw(ff_class* i_this) {
+    cXyz cStack_18;
+
+    mDoLib_project(&i_this->current.pos, &cStack_18);
+    z_check(i_this, &cStack_18);
+    if (i_this->mbNotVisibleZ != 0) {
+        cLib_addCalc0(&i_this->mGlowScale, 1.0f, 0.33333f);
+    } else {
+        cLib_addCalc2(&i_this->mGlowScale, 1.0f, 1.0f, 0.3333f);
+    }
+    fire_fly_draw(i_this);
+    return TRUE;
 }
 
 /* 0000037C-00000D18       .text fire_fly_move__FP8ff_class */
-void fire_fly_move(ff_class*) {
-    /* Nonmatching */
+static void fire_fly_move(ff_class* i_this) {
+    /* Nonmatching - regalloc */
+    s8 cVar1;
+    f32 fVar2;
+    f32 fVar3;
+    f32 dVar7;
+    f32 dVar8;
+    f32 dVar9;
+    f32 fVar10;
+    cXyz local_cc;
+
+    daPy_py_c* player = (daPy_py_c*)dComIfGp_getPlayer(0);
+    dBgS_GndChk chk;
+    i_this->mLiveTimer++;
+    if (i_this->mTimers[2] == 0) {
+        i_this->mTimers[1] = cM_rndF(50.0f) + 40.0f;
+        i_this->mTimers[2] = cM_rndF(200.0f) + 100.0f;
+    }
+    if (i_this->mTimers[1] == 0) {
+        i_this->mScaleTarget = cM_ssin(i_this->mLiveTimer * 1000) * 0.15f * 0.25f + 0.225f;
+    } else {
+        i_this->mScaleTarget = 0.0f;
+    }
+    cLib_addCalc2(&i_this->mScale, i_this->mScaleTarget, 0.1f, 0.05f);
+    cVar1 = i_this->mState;
+    if (cVar1 == 2) {
```

Comment:
If you'd like to fix the matching issue here, the problem is that this is a switch-case on mState, which is why the conditions look so weird here. Take a look this part of the tutorial on identifying and dealing with switch-cases: https://github.com/zeldaret/tww/blob/main/docs/decompiling.md#recognizing-switch-statements